### PR TITLE
planecontrol: add basic plane control unit

### PR DESCRIPTION
**Pull request by HBuczynski**
## Description

Basic implementation of this unit tests control of servos and main motor.

PP-106
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (https://github.com/phoenix-pilot/phoenix-pilot-project/pull/31).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: planecontrol/control.c [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+		signal(SIGTERM, SIG_DFL);
+
+		/* Take terminal control */
+		tcsetpgrp(STDIN_FILENO, getpid());
+
+		plane_run();
+		exit(EXIT_SUCCESS);
+	}
+	else {
+		fprintf(stderr, "planecontrol: vfork failed with code %d\n", pid);
```
Comment (agkaminski) [inline planecontrol/control.c:185]: No need for plane_done and tcsetpgrp on this path 
```suggestion
		fprintf(stderr, "planecontrol: vfork failed with code %d\n", pid);
		return EXIT_FAILURE;
```

File: planecontrol/control.c [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	}
+
+	plane_done();
+
+	/* Take back terminal control */
+	tcsetpgrp(STDIN_FILENO, getpid());
+
+	return EXIT_SUCCESS;
+}
```
Comment (agkaminski) [inline planecontrol/control.c:194]: No new line at the end of the file

File: planecontrol/mma.c [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+struct {
+	handle_t lock;
+} mma_common;
+
+
+int mma_control(float throttle, float proll, float ppitch, float pyaw)
+{
+	int i;
+	float pwm[NUMBER_MOTORS];
```
Comment (agkaminski) [inline planecontrol/mma.c:50]: Nitpick - MOTORS nomenclature doesn't really fit plane's case. Just a remark, maybe control code need some generalization

File: planecontrol/control.c (hunk 1 above)
Comment (HBuczynski) [reply planecontrol/control.c:185]: `plane_init` is invoked by the parent process, so that `proc_done` has to be also invoked by the parent.

**Review by agkaminski — APPROVED** [review]

File: planecontrol/mma.c (hunk 3 above)
Comment (HBuczynski) [reply planecontrol/mma.c]: Yeah, good remark. I have changed name convention.

**Review by niewim19 — COMMENTED** [review]
- Probably accelerometer calibration may be used to clear some body frame dependencies in code, but this needs your feedback.
- Imho mode `sim` may be renamed to `stabilize` as it is common mode name that aims to keep the plane level (It was not testet yet in flight, but in quad-control we had mode `hover` long before it could really hover)

File: planecontrol/control.c [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+	mma_start();
+
+	/* Simulation mode to check control of surfaces
+	 * PIDs are not used. */
+	while (plane_common.mode == mode_sim) {
+		ekf_stateGet(&measure);
+
+		proll = (-measure.roll * 1.6 + M_PI) / (2 * M_PI);
+		ppitch = (-measure.pitch * 1.6 + M_PI / 2) / M_PI;
+		pyaw = (-measure.yaw + M_PI) / (2 * M_PI);
```
[the remark this answers is not in the document]
Comment (HBuczynski) [reply planecontrol/control.c:55]: These equations calculate the value of euler angles in percents. It is not related to initial rotation of the board, it is related to output ranges from EKF:
https://github.com/phoenix-pilot/phoenix-pilot-core/blob/68e086b1846f031dad6c9866c8543855a8678849/ekf/ekflib.h#L24-L26

File: planecontrol/control.c [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+#include <string.h>
+#include <termios.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include <sys/threads.h>
+
+#include <board_config.h>
+
+
+typedef enum { mode_rc = 0, mode_auto, mode_sim } control_mode_t;
```
[the remark this answers is not in the document]
Comment (HBuczynski) [reply planecontrol/control.c]: Done

**Review by niewim19 — APPROVED** [review]
lgtm

**Pull request merged**